packages/loader/container-loader/src/container.ts

```diff
@@ -1628,6 +1674,20 @@ export class Container
 			}
 		}
 
+		// If we have not yet reached `fromSequenceNumber`, we will wait for ops to arrive until we reach it
+		const fromSequenceNumber = loadToSequenceNumber ?? -1;
+		if (this.deltaManager.lastSequenceNumber < fromSequenceNumber) {
+			await new Promise<void>((resolve, reject) => {
+				const opHandler = (message: ISequencedDocumentMessage) => {
+					if (message.sequenceNumber >= fromSequenceNumber) {
```

Comment:
Nit, we can be explicit about the condition rather than using -1

```suggestion
		if (loadToSequenceNumber !== undefined && this.deltaManager.lastSequenceNumber < loadToSequenceNumber) {
			await new Promise<void>((resolve, reject) => {
				const opHandler = (message: ISequencedDocumentMessage) => {
					if (message.sequenceNumber >= loadToSequenceNumber) {
```